airgun/entities/settings.py

```diff
@@ -0,0 +1,53 @@
+from airgun.entities.base import BaseEntity
```

Comment:
Here is my vision of implementation
```
from airgun.entities.base import BaseEntity
from airgun.navigation import NavigateStep, navigator
from airgun.views.settings import SettingEditView, SettingsView


class SettingsEntity(BaseEntity):
    def search(self, value):
        view = self.navigate_to(self, 'All')
        return view.search(value)

    def read(self, property_name):
        """Read settings values"""
        view = self.navigate_to(self, 'All')
        view.search(property_name)
        return view.read()

    def update(self, property_name, value):
        """Update setting property with provided value"""
        view = self.navigate_to(self, 'All')
        view.search(property_name)
        view.table.row()['Value'].widget.fill(value)

@navigator.register(SettingsEntity, 'All')
class ShowAllSettings(NavigateStep):
    """Navigate to All Settings page"""
    VIEW = SettingsView

    def step(self, *args, **kwargs):
        self.view.menu.select('Administer', 'Settings')
```

- Read doesn't need widgets_name as you have only one widget here
- You don't need extra navigation step for Edit as you actually do not navigate anywhere, but stay on 'All'
- You can't just add such `menu...` post actions as they looks completely out of style and redundant. You do `search` for every action, so you get actual data all the time, so not sure why you need it at all. As last resort we can look for refresh, but still don't know why you need it
- We do not work with entities fill like this. Your widget `fill` method should know what you need: that you need click on widget, then fill, then click `submit`, etc. So everything implemented directly in view

Reply:
Excellent @oshtaier!